src/shogun/machine/GaussianProcessMachine.cpp

```diff
@@ -160,6 +165,8 @@ SGVector<float64_t> CGaussianProcessMachine::get_posterior_variances(
 
 	// get kernel matrix and create eigen representation of it
 	SGMatrix<float64_t> k_tsts=kernel->get_kernel_matrix();
+	//To DO in order to speed up the prediction process,
+	//we only need to obtain the diagonal elements of the kernel matrix
```

Comment:
@lambday 
As long as we can speed up the performance of computing diagonal elements of kernel matrix, we can gain the same speed for computing the diagonal elements of gradients wrt hyper-parameter(s).

@karlnapf  your opinion?

However, for now I will implement efficient `get_parameter_gradient_diagonal()` `ONLY for the ARD kernels`.